### PR TITLE
Convert Zombienet (JS) to Monorepo

### LINKED ISSUE #458: Convert ZombieNet to a monorepo
For multiple reasons (e.g. deno support, specific use-cases), its preferable to change ZombieNet to a monorepo, separating `rust` (and crates) and `ts` (and modules).

Proposed structure is:
```
zombienet 
   - src (typescript) 
       - modules (workspaces)
   - crates (rust)
      - crate(workspaces)
   - tests (integration)
   - static-configs (k8s/podman static configs)
   - docs
   - examples
   - admissioncontroller (k8s-support)
```

### PR DESCRIPTION
**Pull request by wirednkod**
This PR (fixes #458) is converting the Zombienet (Javascript part) to a monorepo. It is still a WIP in order to verify that all parts are transfers happened correctly.

The idea is to have in stages the following parts into different modules:
- stage 1: (this PR)
	- orchestrator (the main zombienet part - that will be feasible to be used as a library) -> `@zombienet/orchestrator`
	- utils (all the reusable functions/utils) -> `@zombienet/utils`
	- cli (the command line package -> `@zombienet/zombienet` (for now with the intention to be `zombienet`)
	
- stage 2 (still under consideration):
	- split `test-runner` to a different module

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -20,37 +20,36 @@ jobs:
...
-      run: cargo build --verbose
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-features
 
-    - name: Run tests
-      run: cargo test --verbose --all-targets --all-features
+      - name: Build
+        run: cargo build --verbose
```
Comment (pepoviola) [inline .github/workflows/ci.yml:49]:
```suggestion
        run: cargo build 
```

**Comment by pepoviola** [comment]
We can move the `scripts` directory to the root of the repo

image

And leave the `zombie-wrapper.sh` in the orchestrator since is the only file needed by that module.

Thanks!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (wirednkod) [reply .github/workflows/ci.yml:49]: Thank you. CI will need many changes (e.g. separate CI of JS with cargo and also add directories

File: javascript/packages/orchestrator/package.json [hunk 2]
```diff
@@ -0,0 +1,40 @@
+{
+  "name": "@zombienet/orchestrator",
+  "version": "0.0.1",
+  "description": "ZombieNet aim to be a testing framework for substrate based blockchains, providing a simple cli tool that allow users to spawn and test ephemeral Substrate based networks",
+  "main": "index.js",
+  "types": "index.d.ts",
```
Comment (pepoviola) [inline javascript/packages/orchestrator/package.json:6]:
```suggestion
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
```

File: javascript/packages/cli/src/cli.ts [hunk 3]
```diff
@@ -1,39 +1,41 @@
 #!/usr/bin/env node
+import parser from "@parity/zombienet-dsl-parser-wrapper";
+import { Network, run, start } from "@zombienet/orchestrator/dist";
```
Comment (pepoviola) [inline javascript/packages/cli/src/cli.ts:3]:
```suggestion
import { Network, run, start } from "@zombienet/orchestrator";
```

File: .github/workflows/npm-release.yml [hunk 4]
```diff
@@ -1,24 +1,41 @@
 name: Zombienet NPM Automated Release
 on:
   push:
-    branches: [main]
+    branches: [nik-convert-to-monorepo]
```
Comment (pepoviola) [inline .github/workflows/npm-release.yml:4]: TODO: revert later.

**Review by pepoviola — APPROVED** [review]
awesome job!!

**Pull request merged**